File: go/1/one.go
package one

import (
	"slices"
)

func distanceSum(list1 []int, list2 []int) int {
	slices.Sort(list1)
	slices.Sort(list2)

	distance := 0
	for i, v := range list1 {
		v2 := list2[i]

		if v < v2 {
			distance += v2 - v
		} else {
			distance += v - v2
		}
	}

	return distance
}

File: ts/1/one.ts
export const distanceSum = (arr1, arr2) => {
    const s1 = arr1.sort();
    const s2 = arr2.sort();

    let distance = 0;
    for (let i = 0; i < s1.length; i++) {
        const v1 = s1[i];
        const v2 = s2[i];

        if (v1 < v2) {
            distance += v2 - v1;
        } else {
            distance += v1 - v2;
        }
    }

    return distance
}